src/ecdsa/keys.py

```diff
@@ -923,8 +947,48 @@ def from_der(cls, string, hashfunc=sha1):
         :return: Initialised SigningKey object
         :rtype: SigningKey
         """
-        string = normalise_bytes(string)
-        s, empty = der.remove_sequence(string)
+        s = normalise_bytes(string)
+        curve = None
+
+        # PKCS #8 has the algorithm identifier, including the curve name,
+        # before the actual key. Then it contains the key data within an
+        # octet string.
+        if pkcs8:
+            s, empty = der.remove_sequence(s)
+            if empty != b(""):
+                raise der.UnexpectedDER(
+                    "trailing junk after DER privkey: %s"
+                    % binascii.hexlify(empty)
+                )
+
+            version, s = der.remove_integer(s)
+            if version != 0 and version != 1:
```

Comment:
`version not in (0, 1)` would be terser, wouldn't it?

Reply:
also, I wonder if we shouldn't at least check if presence of `publicKey` matches the `version` (even if we don't parse `publicKey`)

Reply:
I feel like that wouldn't be worthwhile, but I could do it if you feel strongly.

Reply:
regarding `version` or `publicKey`? :)
I don't feel strongly about the latter, the former looks like more readable code to me, so would insist on it